### PR TITLE
Fix #5200 High priority FindBugs warnings

### LINKED ISSUE #5200: Investigate and fix high priority FindBugs warnings
#### Reproduce
- Look at https://molgenis50.gcc.rug.nl/jenkins/job/molgenis-reports/830/findbugsResult/HIGH/ 

#### Expected
- No high priority FindBugs warnings
- molgenis-reports build succeeds

#### Observed
- 28 high priority FindBugs warnings
- molgenis-reports build fails

### PR DESCRIPTION
**Pull request by dennishendriksen**
#### Checklist
- [x] Unit tests

### CODE REVIEW

File: molgenis-data-cache/src/main/java/org/molgenis/data/cache/l1/L1Cache.java [hunk 1]
```diff
@@ -92,6 +92,7 @@ public void evictAll(String entityName)
 	 * @return the retrieved {@link Entity} or Optional.empty() if deletion of this entity is stored in the cache or
 	 * null if no information available about this entity in the cache
 	 */
+	@edu.umd.cs.findbugs.annotations.SuppressWarnings(value = "NP_OPTIONAL_RETURN_NULL", justification = "Intentional behavior")
```
Comment (fdlk) [inline molgenis-data-cache/src/main/java/org/molgenis/data/cache/l1/L1Cache.java]: Ah! zo dus! :)

File: molgenis-data-rest/src/main/java/org/molgenis/data/rest/service/RestService.java [hunk 2]
```diff
@@ -327,7 +327,8 @@ else if (paramValue instanceof String)
 				{
 					throw new MolgenisDataException(
 							format("Attribute [%s] value [%s] does not match date format [%s] or [%s]", attr.getName(),
-									paramStrValue, MolgenisDateFormat.DATEFORMAT_DATE));
+									paramStrValue, MolgenisDateFormat.DATEFORMAT_DATE,
+									MolgenisDateFormat.DATEFORMAT_DATETIME));
```
Comment (fdlk) [inline molgenis-data-rest/src/main/java/org/molgenis/data/rest/service/RestService.java]: But it checks only against the DateFormat, not against DateTime format!

**Comment by fdlk** [comment]
Rest looks fine, but code needs a merge too.

**Comment by dennishendriksen** [comment]
test this please

**Pull request merged**